indico/modules/events/contributions/controllers.py

```diff
@@ -64,6 +66,32 @@ def _checkParams(self, params):
     def _process(self):
         return RH._process(self)
 
+    def _save_custom_fields(self, contrib, custom_field_values):
+        for custom_field in self.event_new.contribution_fields:
+            ContributionFieldValue(contribution_field=custom_field, contribution=contrib,
+                                   data=custom_field_values.get('custom_{}'.format(custom_field.id)))
+
+    def _update_custom_fields(self, contrib, custom_field_values):
+        for custom_field_name, custom_field_value in custom_field_values.iteritems():
+            custom_field_id = int(custom_field_name[7:])  # Remove the 'custom_' part
+            existing_custom_field = contrib.get_field(custom_field_id)
+            if existing_custom_field:
+                if custom_field_value != existing_custom_field.data:
+                    existing_custom_field.data = custom_field_value
+            else:
+                ContributionFieldValue(contribution_field=self.event_new.get_contribution_field(custom_field_id),
+                                       contribution=contrib, data=custom_field_value)
+
+    def _get_field_values(self, form_data):
+        fields = {}
+        custom_fields = {}
+        for name, value in form_data.iteritems():
```

Comment:
I think the improved readability is worth it - the runtime difference is minimal and can be neglected (~2ms depending on the number of entries in the dict)